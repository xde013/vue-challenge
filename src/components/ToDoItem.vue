<template>
  <div class="todo-item">

    <input type='checkbox' v-model="todo.checked">
     {{this.todo.text}}
     <span class='timeago'> - 5 minutes ago</span>
  </div>
</template>

<script>
export default {
  name: 'ToDoItem',
  props: {
    todo: {
      required: true
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="stylus">
    .timeago {
        opacity .5
        font-size .8em
        font-weight 200
    }
</style>
