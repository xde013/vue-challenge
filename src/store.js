import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

export const mutations = {
  addTodo: (state, payload) => {
    state.todos.unshift(payload)
  },
  updateTodo: (state, payload) => {
    // TODO
  },
  removeTodo: (state, payload) => {
    // TODO
  }
}

export const actions = {
  addTodo: ({commit}, payload) => {
    commit('addTodo', payload)
  },
  updateTodo: ({commit}, payload) => {
    commit('updateTodo', payload)
  },
  removeTodo: ({commit}, payload) => {
    commit('removeTodo', payload)
  }
}

export const defaultState = {
  todos: [
    {
      text: 'Think real hard about whats for lunch',
      checked: false,
      createdAt: new Date(Date.now() - 3600)
    },
    {
      text: 'Buy some Pantos and start looking for Lambos',
      checked: false,
      createdAt: new Date(Date.now() - 7200)
    }
  ]
}

export default new Vuex.Store({
  state: defaultState,
  mutations,
  actions
})
