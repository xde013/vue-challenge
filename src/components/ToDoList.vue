<template>
  <div class="todo-list">

      <to-do-input class='input'></to-do-input>

      <div class='todo-items'>
          <to-do-item v-for="todo in $store.state.todos" :key="todo.createdAt.getTime()" :todo="todo"></to-do-item>
      </div>
  </div>
</template>

<script>
import ToDoInput from './ToDoInput'
import ToDoItem from './ToDoItem'
export default {
  name: 'ToDoList',
  components: {ToDoInput, ToDoItem}
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="stylus">
    .todo-items {
        margin-top 30px
        display inline-block
        text-align left
    }
</style>
