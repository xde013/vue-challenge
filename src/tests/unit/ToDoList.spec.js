import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import ToDoList from '@/components/ToDoList.vue'
import ToDoItem from '@/components/ToDoItem.vue'
import { defaultState, getters, mutations, actions } from '@/store'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('ToDoList.vue', () => {
  let store
  let localConfig
  beforeEach(() => {
    store = new Vuex.Store({
      state: defaultState,
      getters: getters,
      mutations: mutations,
      actions: actions
    })
    localConfig = { store, localVue }
  })

  it('renders todo-item for each item in todos', () => {
    const wrapper = shallowMount(ToDoList, localConfig)
    expect(wrapper.findAll(ToDoItem)).toHaveLength(store.state.todos.length)
  })

  it('matches snapshot', () => {
    const wrapper = shallowMount(ToDoList, localConfig)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
