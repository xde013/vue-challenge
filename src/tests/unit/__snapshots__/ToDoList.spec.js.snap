// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoList.vue matches snapshot 1`] = `
<div class="todo-list">
  <todoinput-stub class="input"></todoinput-stub>
  <div class="todo-items">
    <todoitem-stub todo="[object Object]"></todoitem-stub>
    <todoitem-stub todo="[object Object]"></todoitem-stub>
  </div>
</div>
`;
